pipeline/scripts/ci/upstream_cli.py

```diff
@@ -0,0 +1,159 @@
+"""
+This script provides the upstream latest rpm repo path and container image.
+Please note that this script should be run under root privilages
+inorder to pull the image and podman should be installed on that host.
+"""
+import fcntl
+import logging
+import os
+import re
+import sys
+import time
+
+import requests
+import yaml
+from docopt import docopt
+
+LOG = logging.getLogger(__name__)
+
+
+def store_in_upstream(branch_name, repo, image, version):
+    """Method to update it in upstream.yaml.
+    Args:
+        branch_name: Name of upstream branch
+        repo: upstream branch compose URL
+        image: upstream branch image
+        version: upstream branch ceph version.
+    """
+    file_name = "/ceph/cephci-jenkins/latest-rhceph-container-info/upstream.yaml"
+    stream = open(file_name, "r")
+    repo_details = yaml.load(stream)
+    # To handle new release which is not updated yet
+    if repo_details[branch_name]:
+        existing_version = repo_details[branch_name]["ceph-version"]
+    else:
+        raise Exception(f"{branch_name} not found in upstream.yaml")
+    # update only if current version is greater than existing version.
+    if version > existing_version:
+        while True:
```

Comment:
This might end in a never ending loop. Kindly include a timeout

Reply:
@psathyan No It's not a never ending loop 
case 1) if it's first time job access the file perform the write and there is Break after closing file so It comes out from while.
case 2) If some job doing some write operation in that file, another job will get exception in opening file as per **fcntl lock** so it will wait for 2 minutes before retrying to access that file once First process finishes it;s write operation second process keeps waiting and retrying.

Reply:
Keeping a timeout would avoid the scenario wherein the lock is held indefinitely by an application. A failure would allow us to be notified and take appropriate steps.

Reply:
@psathyan Thanks for considering deadlock case handled that with 10 minutes timeout,
Please review.